src/ctapipe/calib/camera/extractor.py

```diff
@@ -0,0 +1,426 @@
+    charge_median_cut_outliers = List(
+        [-0.3, 0.3],
+        help="Interval of accepted charge values (fraction with respect to camera median value)",
+    ).tag(config=True)
+    charge_std_cut_outliers = List(
+        [-3, 3],
+        help="Interval (number of std) of accepted charge standard deviation around camera median value",
+    ).tag(config=True)
+    time_cut_outliers = List(
+        [0, 60], help="Interval (in waveform samples) of accepted time values"
+    ).tag(config=True)
+    sigma_clipping_max_sigma = Int(
+        default_value=4,
+        help="Maximal value for the sigma clipping outlier removal",
+    ).tag(config=True)
+    sigma_clipping_iterations = Int(
+        default_value=5,
+        help="Number of iterations for the sigma clipping outlier removal",
+    ).tag(config=True)
+
+    def __init__(self, subarray, config=None, parent=None, **kwargs):
+        """
+        Parameters
+        ----------
+        subarray: ctapipe.instrument.SubarrayDescription
+            Description of the subarray
+        config : traitlets.loader.Config
+            Configuration specified by config file or cmdline arguments.
+            Used to set traitlet values.
+            Set to None if no configuration to pass.
+        kwargs
+        """
+        super().__init__(config=config, parent=parent, subarray=subarray, **kwargs)
+
+        # Check validity of sample size and update frequency
+        if self.sample_size < self.update_frequency:
+            raise ValueError(
+                f"The 'update_frequency' ({self.update_frequency}) must be less than or equal to the 'sample_size' ({self.sample_size})."
+            )
+        # set up the buffer
+        self.n_events_in_buffer = 0  # number of events in sample
+        self.trigger_time = []  # trigger time of event in sample
+        self.charges = []  # charge per event in sample
+        self.arrival_times = []  # arrival time per event in sample
+        self.broken_pixels = []  # masked pixels per event in sample
+
+        # load the waveform charge extractor
+        self.extractor = ImageExtractor.from_name(
+            self.image_extractor_type, parent=self, subarray=subarray
+        )
+
+    def __call__(self, event: ArrayEventContainer) -> Bool:
+        """
+        Extract the charge and the time from a calibration event.
+        Process sample if enough statistics is reached and
+        clean up the buffer.
+
+        Parameters
+        ----------
+        event : ArrayEventContainer
+
+        Returns: True if the MonitoringCameraContainer is updated,
+                 False otherwise
+        """
+
+        # extract the charge of the event and
+        # the peak position (assumed as time for the moment)
+        dl1: DL1CameraContainer = self._extract_charge_and_time(event)
+
+        # append valid dl1 events
+        self.n_events_in_buffer += 1
+        self.trigger_times.append(event.trigger.tel[self.tel_id].time)
```

Comment:
`self.tel_id` seems not to be defined anymore.

Also: these components should be able to handle multiple telescopes I think, so we keep e.g. a dict from tel_id to the properties.